readthedocs/config/config.py

```diff
@@ -806,6 +815,30 @@ def validate_sphinx(self):
 
         return sphinx
 
+    def validate_final_doc_type(self):
+        """
+        Validates that the doctype is the same as the admin panel.
+
+        This a temporal validation, as the configuration file
+        should support per version doctype, but we need to
+        adapt the rtd code for that.
+        """
+        dashboard_doctype = self.defaults.get('doctype', 'sphinx')
+        if self.doctype != dashboard_doctype:
+            key = 'mkdocs' if self.doctype == 'mkdocs' else 'sphinx'
+            needed_doctype = (
+                'mkdocs' if dashboard_doctype == 'mkdocs' else 'sphinx'
+            )
+            self.error(
+                key,
+                'Your project is configured as "{doctype}" in your admin '
+                'dashboard, but there is no "{key}" key specified.'.format(
+                    doctype=dashboard_doctype,
+                    key=needed_doctype,
```

Comment:
I think what you meant was:

```python
    doctype=needed_doctype,
    key=key,
```

Reply:
This was on purpose, as we say that the project is configured as `sphinx_htmldir` in the admin dashboard, and the user is missing a `sphinx` key in their config file.

`key` represents the current doctype from the config file.

Dashboard -> sphinx
Config -> mkdocs

Error: Your project is configured as "sphinx" in your admin dashboard, but there is no "sphinx" key specified.

Reply:
so, in the previous example, key would be `mkdocs` We can put the key as `.` if that's looks confusing

Reply:
Also, we need the dashboard_doctype, otherwise we will have some weird errors like:

Dashboard -> sphinx_htmldir
Config -> sphinx.html

 `Your project is configured as "sphinx" in your admin dashboard, but there is no "sphinx" key specified`


The current message for this case looks like:

 `Your project is configured as "sphinx_htmldir" in your admin dashboard, but there is no "sphinx" key specified`

Isn't perfect neither... but better than the above p:

Reply:
What about `Your project is configured as "sphinx_htmldir" in your admin dashboard, it doesn't match the documentation type from the configuration`

Reply:
> Isn't perfect neither... but better than the above p:

Exposing our internal slug to users isn't great UX, so i actually prefer the first. I wanted to originally use the verbose option name, but this is maybe annoying to get from the choice list -- for example, if i was using the model, i'd use the `get_documentation_type_display()` automatic function.

The `_get_FIELD_display` method is very small, we can likely replicate it very easily:
https://github.com/django/django/blob/cc79c7ee637e65c8da27e56d746c87903d5ec901/django/db/models/base.py#L902-L904

That way, we give the user an error that matches what the *user* sees in our admin, not what *we see in our database*. So, ideally:

```
Your project is configured as Sphinx HTMLdir in your admin dashboard,
but there is no "sphinx" key specified
```